internal/server/middleware.go

```diff
@@ -182,3 +182,76 @@ func getCookie(req *http.Request, name string) (string, error) {
 	}
 	return url.QueryUnescape(cookie.Value)
 }
+
+func OrganizationFromDomain(defaultOrgName, defaultOrgDomain string) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		db := getDB(c)
+		host := c.Request.Header.Get("Host")
+
+		// remove port
+		if i := strings.Index(host, ":"); i >= 0 {
+			host = host[0:i]
+		}
+
+		if len(host) == 0 {
+			// tests are lazy and don't set Host
+			host = defaultOrgDomain
+		}
+
+		if len(host) > 0 {
+			org, err := data.GetOrganization(db, data.ByDomain(host))
+			if err != nil {
+				if errors.Is(err, internal.ErrNotFound) {
+					org, err = getDefaultOrg(db, defaultOrgName, defaultOrgDomain)
+					if err != nil {
+						sendAPIError(c, fmt.Errorf("creating default organization: %w", err))
+						return
+					}
+					goto success
+				} else {
+					logging.Errorf("fetching org: %s", err)
+				}
+				// failed to find org
+				c.Next()
```

Comment:
Failing to find the org feels like it should fail, what would happen without the org specified?

Reply:
There are separate middleware for finding the organization and requiring the organization. I thought maybe some unauthenticated endpoints should work properly with or without an organization, but I'm not 100% true that's the case. Anyway the separation lets you load the org without failing when it's missing. If we don't need it we should combine the middleware